pex/vendor/__main__.py

```diff
@@ -189,8 +189,22 @@ class VendorizeError(Exception):
 
 def vendorize(root_dir, vendor_specs, prefix):
   for vendor_spec in vendor_specs:
-    cmd = ['pip', 'install', '--upgrade', '--no-compile', '--target', vendor_spec.target_dir,
-           vendor_spec.requirement]
+    # NB: We set --no-build-isolation to prevent pip from installing the requirements listed in
```

Comment:
I'm going to see if this can be reverted bythe expedient of removing the install metadata we don't need (`entry_points.txt`, `INSTALLER`,  and `WHEEL`). That should allow reverting these changes as well as the patch to pip that was needed in https://github.com/pantsbuild/pip/commit/e2553a16ba0e0272a915ceff848a29d27a088e83